sdk/tests/utilities/id_generator.py

```diff
@@ -0,0 +1,62 @@
+import uuid
+
+
+class IdGenerator:
+
+    def __init__(self, default_scope=None):
+        """
+
+        Parameters
+        ----------
+        default_scope   scope to use for subsequent generate calls when do scope is provided
+        """
+        self.default_scope = default_scope if default_scope is not None else "sdk_example"
```

Comment:
Is default scope the right name for the attribute? It is default when it is `sdk_example` but otherwise it is arguably `set scope`.

The test `test_uses_provided_scope` makes this point a little more clearly.

Reply:
fair comment, how about setting the class level var to just `scope`?

```
 def __init__(self, default_scope=None):
        self.scope = default_scope if default_scope is not None else "sdk_example"
```

Reply:
that is actually probably better!